backend/src/main/java/com/google/capstone/Favorite.java

```diff
@@ -0,0 +1,38 @@
+package com.google.capstone;
+
+import com.google.appengine.api.datastore.Entity;
+
+/**
+ * Datatype to represent a favorite object.
+ */
+public class Favorite {
+  private final String uid;
+  private final String collection;
+  private final String artworkId;
+  private final String name;
+  private final String alt;
+  private final String url;
+  private final long timestamp;
+
+  public Favorite(String uid, String collection, String artworkId,
+                  String name, String alt, String url, long timestamp) {
+    this.uid = uid;
+    this.collection = collection;
+    this.artworkId = artworkId;
+    this.name = name;
+    this.alt = alt;
+    this.url = url;
```

Comment:
These three fields seem to be redundant. If so, it might cause [update anomaly](https://en.wikipedia.org/wiki/Database_normalization). Normally we get these informations from Chicago API/our own database? Is it possible that name/alt/url get updated later?

>update anomaly: The same information can be expressed on multiple rows; therefore updates to the relation may result in logical inconsistencies.

Reply:
I did consider this, but there a lot that needs to be setup for me to to get it directly for the previews in the gallery view of favorites and it was easier not to. The close up card however will get it directly from the appropriate API. I planned on changing this if time permits in a new PR.